src/Search.php

```diff
@@ -4620,6 +4620,22 @@ public static function addWhere($link, $nott, $itemtype, $ID, $searchtype, $val,
                     break;
             }
         }
+
+        // In some cases, we will need to replace the WHERE condition by a subquery.
+        // This is needed to avoid false positif for  for negative conditions on
+        // children items (e.g. search for tickets that DON'T have a given group)
+        // See https://github.com/glpi-project/glpi/pull/13684 for detailed examples
+        $use_where_subquery = false;
+
+        // Special case when having a negative condition on children items with
+        // the "contains" or "not contains" search type.
+        // The subquery will need to be constructed differently
+        $use_where_subquery_on_name = false;
```

Comment:
This is related to a "text" search, but not specific to `name` field.
```suggestion
        $use_subquery_on_text_search = false;
```